cranelift/filetests/src/function_runner.rs

```diff
@@ -114,7 +114,7 @@ impl TestFileCompiler {
             unsafe extern "C" {
                 safe fn cosf(f: f32) -> f32;
             }
-            let f = 1.2_f32;
+            let f = std::hint::black_box(1.2_f32);
```

Comment:
Digging into this, it looks like LLVM sees in the assertion below `assert_eq!(cosf(x), cosf(x))` and then LLVM has built-in knowledge that `cosf` is pure so it's optimized out entirely. That means libm doesn't get linked at all, meaning later `dlsym` calls fail.

Using `black_box` here is apparently enough, but unsure how long it'll last necessarily.